### PR TITLE
feat: add failure workflow step to report issues

### LINKED ISSUE #352: Create issue on GitHub Actions failure
**Description**

We should have visibility into Actions failures. For example, if the snapshot/timestamp action failed to trigger, there is currently no visibility into this failure, and it would cause an outage for verifiers. Assuming this is supported by GHA, I propose filing an issue on failure and tagging the maintainers.

cc @asraa

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

Inspired by https://github.com/slsa-framework/example-package/blob/main/.github/workflows/scripts/e2e-report-failure.sh

Fixes https://github.com/sigstore/root-signing/issues/352

### CODE REVIEW

**Review by cpanato — COMMENTED** [review]
lgtm

File: .github/workflows/scripts/report-failure.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+Date: $RUN_DATE
+EOF
+}
+
+THIS_FILE=$(this_file)
+create_issue_body
+
+ISSUE_ID=$(gh -R "$ISSUE_REPOSITORY" issue list --label "type:bug" --state open -S "$THIS_FILE" --json number | jq '.[0]' | jq -r '.number' | jq 'select (.!=null)')
+
+if [[ -z "$ISSUE_ID" ]]; then
```
Comment (Hayden-IO) [inline .github/workflows/scripts/report-failure.sh:47]: Deduping! This is awesome.

**Review by Hayden-IO — DISMISSED** [review]

**Comment by Hayden-IO** [comment]
This looks great!

File: .github/workflows/snapshot-timestamp.yml [hunk 2]
```diff
@@ -83,6 +83,16 @@ jobs:
           path: repository/repository/
           retention-days: 5
 
+  if-failed:
+    runs-on: ubuntu-latest
+    needs: [snapshot_and_timestamp]
+    if: needs.snapshot_and_timestamp.result == 'failure'
```
Comment (Hayden-IO) [inline .github/workflows/snapshot-timestamp.yml:95]: Does the token need any other permissions to create issues?

Comment (asraa) [reply .github/workflows/snapshot-timestamp.yml:95]: The workflow has GITHUB_TOKEN. that has perms

Comment (cpanato) [reply .github/workflows/snapshot-timestamp.yml:95]: looking at it again the workflow only has the 
```
    permissions:
      id-token: 'write'
```

we need to add the issues permission as well

Comment (asraa) [reply .github/workflows/snapshot-timestamp.yml:95]: Ah, I guess that's per-job permissions. The snapshot-timestamp job has those, i'll make it explicit that the job permission for the failures have issue writes. thank you!

**Review by Hayden-IO — APPROVED** [review]

**Pull request merged**